exec/java-exec/src/main/java/org/apache/drill/exec/planner/cost/DrillRelMdMaxRowCount.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.planner.cost;
+
+import org.apache.calcite.rel.core.TableScan;
+import org.apache.calcite.rel.metadata.ReflectiveRelMetadataProvider;
+import org.apache.calcite.rel.metadata.RelMdMaxRowCount;
+import org.apache.calcite.rel.metadata.RelMetadataProvider;
+import org.apache.calcite.rel.metadata.RelMetadataQuery;
+import org.apache.calcite.util.BuiltInMethod;
+import org.apache.drill.exec.planner.logical.DrillLimitRel;
+
+public class DrillRelMdMaxRowCount extends RelMdMaxRowCount {
+
+  private static final DrillRelMdMaxRowCount INSTANCE = new DrillRelMdMaxRowCount();
+
+  public static final RelMetadataProvider SOURCE = ReflectiveRelMetadataProvider.reflectiveSource(BuiltInMethod.MAX_ROW_COUNT.method, INSTANCE);
+
+  public Double getMaxRowCount(DrillLimitRel rel, RelMetadataQuery mq) {
+    return rel.getRows();
+  }
+
+  @Override
+  public Double getMaxRowCount(TableScan rel, RelMetadataQuery mq) {
+    return rel.getRows();
```

Comment:
Sorry, I had been meaning to reply sooner.  I think overloading the getMaxRowCount() to return rel.getRows() can create potential issues...because getMaxRowCount() should always return whatever is the *maximum* row count possible for that RelNode.  Here, if you return TableScan.getRows(), the value is an *estimate*, which means in reality it could be higher.   The caller might make incorrect decision based on this value.  

I am thinking about your original motivation for the changes.  Are you materializing the results into a single-row table ?  It sounds like you want a special table scan whose max row count is 1.    Is materializing the only option ?   (the reason I am curious is it is odd to materialize very small data sets such as 1 row).

Reply:
@amansinha100  Your assumption is right. My initial motivation is to materialize the count distinct value of one table column without the group by clause into a single-row table i.e. a cache.  To accelerate users' query time, this materialization seems reasonable. 

Since getMaxRowCount() only invoked by the `JoinUtils.isScalarSubquery` ,there's no potential issues.